File: libnetcache/src/main/java/com/skymajo/libnetcache/UrlCreator.java
package com.skymajo.libnetcache;

import java.io.UnsupportedEncodingException;
import java.net.URLEncoder;
import java.util.Map;

public class UrlCreator {
    public static String createUrlFormParams(String url, Map<String , Object> params){
        StringBuilder builder = new StringBuilder();
        builder.append(url);
        if (url.indexOf("?")>0 || url.indexOf("&")>0){
            builder.append("&");
        }else{
            builder.append("?");
        }
        for (Map.Entry<String,Object>entry:params.entrySet()){
            String value = null;
            try {
                value = URLEncoder.encode(String.valueOf(entry.getValue()),"UTF-8");
            } catch (UnsupportedEncodingException e) {
                e.printStackTrace();
            }
            builder.append(entry.getKey()).append("=").append(value).append("&");
        }
        builder.deleteCharAt(builder.length()-1);
        return builder.toString();
    }
}

File: libnetcache/src/main/java/com/skymajo/libnetcache/UrlCreator.kt
package com.skymajo.libnetcache

import android.util.Log
import java.lang.StringBuilder
import java.net.URLEncoder

object UrlCreator {
    fun creatUrlFromParams(url:String,params:Map<String,Any>):String{
        if( params.isEmpty()) return url
        var builder= StringBuilder()
        builder.append(url)
        Log.e("UrlCreator","${url.indexOf("?")>0 || url.indexOf("&") >0}")
        if(url.indexOf("?")>0 || url.indexOf("&") >0){
            Log.e("UrlCreator","")
            builder.append("&")
        }else {
            builder.append("?")
        }
        for((key,value) in params){
            var encodeValue=URLEncoder.encode(value.toString(),"UTF-8")
            builder.append(key).append("=").append(encodeValue).append("&")

        }
        builder.deleteCharAt(builder.length-1)
        Log.e("UrlCreator","${builder.toString()}")
        return builder.toString()
    }
}